course3/homework-string/my_string.cc

```diff
@@ -0,0 +1,63 @@
+#include "my_string.h"
+#include <cstring>
+#include <stdexcept>
+
+MyString::MyString(const char* str) : buf(std::make_unique<char[]>(std::strlen(str)+1)) {
+	std::strcpy(buf.get(), str);
+}
+
+MyString::MyString(const MyString& obj) : buf(std::make_unique<char[]>(obj.size()+1)) {
+	std::strcpy(buf.get(), obj.buf.get());
+}
+
+bool MyString::operator==(const MyString& rhs) const {
+	if (!rhs.buf xor !buf)
+		return false;
+	if (!rhs.buf and !buf)
```

Comment:
I strongly advise against using the text form of logical operators.
Not all compilers support them (*cough*VS*cough*), and they are bitwise operatrors, not boolean operators (```&``` vs ```&&```), so they can produce surprising results at times.